beta-src/src/components/ui/WDUnitController.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from "react";
+import { gameIconProps } from "../../interfaces/Icons";
+import UIState from "../../enums/UIState";
+import debounce from "../../utils/debounce";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import {
+  gameApiSliceActions,
+  gameData,
+  gameOrder,
+} from "../../state/game/game-api-slice";
+
+interface UnitControllerProps {
+  meta: gameIconProps["meta"];
+  setIconState: React.Dispatch<UIState>;
+}
+
+const WDUnitController: React.FC<UnitControllerProps> = function ({
+  children,
+  setIconState,
+  meta,
+}): React.ReactElement {
+  const dispatch = useAppDispatch();
+
+  const commands = useAppSelector(
+    (state) => state.game.commands.unitCommands[meta.unit.id],
+  );
+
+  const { data } = useAppSelector(gameData);
+
+  const order = useAppSelector(gameOrder);
+
+  if (order.unitID === meta.unit.id) {
+    setIconState(UIState.SELECTED);
+  } else {
+    setIconState(UIState.NONE);
+  }
+
+  if (commands && commands.size > 0) {
+    const firstCommand = commands.entries().next().value;
+    if (firstCommand) {
+      const [key, value] = firstCommand;
+      switch (value.command) {
+        case "HOLD":
+          setIconState(UIState.HOLD);
+          dispatch(
+            gameApiSliceActions.deleteCommand({
+              type: "unitCommands",
+              id: meta.unit.id,
+              command: key,
+            }),
+          );
+          break;
+        default:
+          break;
+      }
+    }
```

Comment:
This logic is very similar to the one in WDTerritory.tsx. Can we move this to a utility file and handle this in 1 switch check for all the cases.

Reply:
Hmm, I originally tried to do this but `useAppDispatch` is unusable in a utility function since it is a redux react hook. However, I didn't try passing a callback as a function argument. I can try that and see how it goes.

Reply:
I did find a better way, thanks.